### DISCUSSION TITLE
Change compassViewPosition to type string, instead of number...

### DISCUSSION THREAD

**Discussion by thispastwinter** [rnmapbox/maps#1569] [Ideas]
Currently compassViewPosition accepts a number range from 0-4. I don't feel this is very intuitive. I think it makes more sense to just have 4 string types.

Example:

`type CompassViewPosition = "topLeft" | "topRight" | "bottomLeft" | "bottomRight"`

What this would then look like: 

```
<MapboxGL.MapView
   compassViewPosition="topLeft"
   style={styles.mapBox}
   zoomEnabled
   >
   {children}
 </MapboxGl.MapView>
```

Apologies for not necessarily following the template to a "t", as this isn't really a bug, but a preference over the current api. I'm also happy to put up a PR if anyone else agrees with this.

**Comment by thispastwinter** [comment]
Thanks for appropriately categorizing this! Shouldn't have filed this in issues.

**Reply by ferdicus** [reply]
all good 👍🏿 
sounds fine to me

If you want to be real fancy you could either add a completely new prop which deals with that, make the current prop deal with both. 

I leave it up to you. 
In general, always feel free to open a PR :)

Cheers 🍻
